### ISSUE TITLE
Post-launch Issues With BDP Docs

### ISSUE THREAD

**Issue by laurenrother** [basho/basho_docs#1796]
- [x] Every link is broken.

**Docs content issues**
- [x] Java install instructions for CentOS are incorrect in three places (see 3 separate comments below).
- [x] Setting Up A Cluster has several incorrect commands (see comment below).
- [x] Some Markdown weirdness @ top of Getting Started with Cache Proxy (see more in comment below).
- [x] Incorrect command in Getting Started with Cache Proxy (see comment below).
- [x] See https://github.com/basho/basho_docs/pull/1795

**Site build/configuration issues**
- [x] Downloads link goes to Riak KV.
- [x] Learn About Data Platform index.md is . . . not showing up at all.
- [x] Using Data Platform index.md page is also not showing up at all.
- [x] Graphic for BDP not showing up on main docs page.
- [x] Images on http://docs.basho.com/dataplatform/latest/learn-about-dataplatform/cache-proxy-features/ not rendering.
- [ ] Cannot link to headers smaller than h2 at all.
- [ ] Cannot link to headers in any doc that does not have a TOC specified in frontmatter.
- [ ] All links, headers and not-headers that are not explicitly spelled out with a full URL resolve to riak/latest. Reason unkown
- [ ] Font sizes for h2 and h3 are too close (see comment below).

**Comment by paegun** [comment]
[ ]  Java install truncated here http://docs.basho.com/dataplatform/latest/installing/
wget only gets the file, it needs to be installed using yum, add `sudo yum -y localinstall jdk-8u45-linux-x64.rpm`

**Comment by ph07** [comment]
* Font sizes - it seems second and third level use the same font type and size. First level heading is the same size as page title. Looking at Riak articles, like http://docs.basho.com/riak/latest/upgrade-v20, I see the headings differ better in sizes.

**Comment by paegun** [comment]
[ ] http://docs.basho.com/dataplatform/latest/using-dataplatform/configuration/setup-a-cluster/ 
$ {JAVA_HOME}/bin/java -version
should read
$ ${JAVA_HOME}/bin/java -version

**Comment by ph07** [comment]
Page: https://github.com/basho/basho_docs/blob/dataplatform/1.0.0/source/languages/en/dataplatform/installing.md, section 'Java 8', Java 8, inside shell code block
Now:
```
$ 
sudo apt-get install software-properties-common
sudo apt-get install python-software-properties
...
```
Should be
```
$ sudo apt-get install software-properties-common
$ sudo apt-get install python-software-properties
...
```

**Comment by ph07** [comment]
Page: https://github.com/basho/basho_docs/blob/dataplatform/1.0.0/source/languages/en/dataplatform/installing.md, section 'Java 8', Java 8, inside 3rd shell code block
Now:
```
grep JAVA_HOME /etc/environment >/dev/null 2>&1 || test $? -ne 0 && sudo bash -c "echo JAVA_HOME=$(dirname $(dirname $(readlink -f $(which javac)))) >>/etc/environment"
export JAVA_HOME
```
Should be
```
$ grep JAVA_HOME /etc/environment >/dev/null 2>&1 || test $? -ne 0 && sudo bash -c "echo JAVA_HOME=$(dirname $(dirname $(readlink -f $(which javac)))) >>/etc/environment"
export JAVA_HOME
```

**Comment by korry8911** [comment]
on http://docs.basho.com/dataplatform/1.0.0/installing/
not really an issue but 

Unpack the packages using sudo dpkg -i for each one.

might be more clear if 

Unpack the packages using "sudo dpkg -i package_name_here" for each one.

**Comment by ph07** [comment]
https://github.com/basho/basho_docs/blob/dataplatform/1.0.0/source/languages/en/dataplatform/installing.md

This part 
> Enterprise Note: If you are an Enterprise customer, make sure to download the -extras package as well. The -extras package is available alongside the regular packages in the usual Zendesk forums.

is not exactly correct. All customers, both EE and OSS should download both main and extras packages. Might be good to note, that EE customers **should** download EE version from Zendesk page or contact their sales manager to request a trial.

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/1.0.0/using-dataplatform/configuration/setup-a-cluster/

> Note on Amazon Web Services
> AWS security profile must allow incoming and outgoing traffic from ip/ports used by Riak, Spark, and BDP. A list of default ports is at the end of this document.

Please note this the list below lists default ports only for Spark and BDP, but not Riak itself. Those are the ports I know, please verify with engineering if more ports need to be added. Additionally, there might be some port ranges, used by clients - need to get advise from Riak team. 
```
EPMD listener = 4369/tcp
PBC (ProtoBuf) = 8087/tcp
HTTP = 8098/tcp
Riak Handoff = 8099/tcp
Solr = 8093
Solr JMX = 8985
Leader Election = 5323 (Enterprise Edition only)
Cluster Manager = 9080
Riak EE JMX = 41110 (Enterprise Edition only)
```

**Comment by ph07** [comment]
https://github.com/basho/basho_docs/blob/dataplatform/1.0.0/source/languages/en/dataplatform/upgrading.md

* Upgrading word appears twice. The reason is using H1 header and page title. I think other pages do not use H1 header on top.
* Consider removing or shortening main header - it renders ugly at this size

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/latest/release-notes/#Installing

Question - do we have a special language for Enterprise customers? They **should not** get the files there as those are for OSS users, but from the Zendesk page

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/1.0.0/using-dataplatform/configuration/replace-spark-cluster-manager/

This page does not render well. I proposed some formatting changes as separate PR, but it needs more work, e.g. numbered lists do not work inside the box ('If you see a hostname rather..'), numbered lists gets restarted, so multiple '1.' Maybe just remove them?

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/1.0.0/learn-about-dataplatform/cache-proxy-features/

* Link is not rendered and broken - 'Cache proxy is available to [Enterprise users only][ee].'
* Diagrams (images) are not shown. Are those links correct?
[readthrough-strategy]: ./images/readthrough-strategy.png
[writethrough-sequence]: ./images/writethrough-sequence.png
* Afaik, Write Through is coming in the next release - but please verify with Eng team

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/1.0.0/using-dataplatform/configuration/replace-spark-cluster-manager/

Should it be marked as 'Enterprise only' ? Please check with Eng team.

**Comment by ph07** [comment]
IMPORTANT - IMHO, we need "Using Spark with BDP" instructions. Setting up Spark Cluster does not explain how to use Spark with Riak (at least basic operations), how to add bundled Spark Connector to their app, how to reference it in Java/Scala code

**Comment by TheOnlyRew** [comment]
Responding to numerous comments above...

>https://github.com/basho/basho_docs/blob/dataplatform/1.0.0/source/languages/en/dataplatform/upgrading.md

> * Upgrading word appears twice. The reason is using H1 header and page title. I think other pages do not use H1 header on top.
> * Consider removing or shortening main header - it renders ugly at this size

In this case, I believe the `title: ` metadata should be changed from `Upgrading` to `Upgrading from Basho Data Platform Beta to 1.0.0`, and the header should be removed. No need to mess with font sizes.
This was changed in #1795, but I disagree with that PR.

>http://docs.basho.com/dataplatform/latest/release-notes/#Installing

> Question - do we have a special language for Enterprise customers? They should not get the files there as those are for OSS users, but from the Zendesk page

It's my understanding that Enterprise customers will be given login credentials to the access-controlled Zendesk forums that host the EE packages and EE install instructions. The Docs historically do not include installation instructions for -- or even references to -- enterprise packages, and I believe we should maintain that consistency.

> http://docs.basho.com/dataplatform/1.0.0/using-dataplatform/configuration/replace-spark-cluster-manager/

> This page does not render well. I proposed some formatting changes as separate PR, but it needs more work, e.g. numbered lists do not work inside the box ('If you see a hostname rather..'), numbered lists gets restarted, so multiple '1.' Maybe just remove them?

This conversation should be continued in #1804. If there are additional changes to be made for that PR, please update the `ph07-patch-6` branch.

-----

@ph07, in the future, please chose descriptive names for your commits and PRs. Five of your last eight PRs are all named "Minor formatting changes" which makes it difficult to talk about them individually, especially because they are numerically sequential. Similarly, please chose descriptive commit messages for your changes. Seeing repetitions of "Minor formatting changes" or "Update replace-spark-cluster-manager.md" in the git logs won't do anyone any favors when we need to review previous changes.

**Comment by TheOnlyRew** [comment]
Using ```` ```shell```` as an opener for code-blocks that should have bash-style syntax highlighting won't work. The div's will be correctly labeled with "Shell", but the syntax highlighting will likely be incorrect. Those openers should be changed to ```` ```bash```` so we get both the "Shell" label and correct syntax highlighting, and we maintain consistency with the rest of the documentation site.

**Comment by TheOnlyRew** [comment]
In both using-dataplatform/configuration/setup-a-cluster.md and learn-about-dataplatform/leader-election-service.md we're using ```` ```riak.conf```` to open code blocks. These should be ```` ```riakconf```` to insure syntax highlighting is correct, and to remain consistent with the rest of the docs site.

**Comment by laurenrother** [comment]
@ph07, in addition to @TheOnlyRew's comments above:
>http://docs.basho.com/dataplatform/latest/release-notes/#Installing
>
>Question - do we have a special language for Enterprise customers? They should not get the files there as those are for OSS users, but from the Zendesk page

As Drew mentioned, we generally do not include Enterprise installation instructions on the Docs site. I have done so in this case because there is a little odd wiggling given the beta program, and I wanted to make sure every user is where they need to be. All mention of Enterprise on the install page will disappear with the next x-version release.

>http://docs.basho.com/dataplatform/1.0.0/using-dataplatform/configuration/replace-spark-cluster-manager/
>
>This page does not render well. I proposed some formatting changes as separate PR, but it needs more work, e.g. numbered lists do not work inside the box ('If you see a hostname rather..'), numbered lists gets restarted, so multiple '1.' Maybe just remove them?

Markdown, in its infinite wisdom, does not support inserting code examples or pictures or anything in between list items or steps. This is done, ostensibly, to encourage simplicity... which is great when the thing you are documenting functions simply. It is less than ideal when it does not. There are workarounds for rendering. I tried some workarounds with this release to no avail. We're working on a solution. I think steps are important for readability and ease of use. If we can't find a workaround for rending interrupted steps, I'll switch to the more annoying and less elegant **Step 1.**, **Step 2.**, etc.

>IMPORTANT - IMHO, we need "Using Spark with BDP" instructions. Setting up Spark Cluster does not explain how to use Spark with Riak (at least basic operations), how to add bundled Spark Connector to their app, how to reference it in Java/Scala code

I asked for these in predocs, but was given the replacement stuff. Since we're in-progress with those, I'll consider that one fixed, too :)

**Comment by laurenrother** [comment]
The h3 and below anchor issue is being worked on in a much longer term project. Everything else has been resolved. As such, I am closing this ticket. Further issues with the docs can be filed in individual issues/tickets. :)

**Comment by ph07** [comment]
@laurenrother - great job on resolving those remarks. One was missed - list of default ports.

**Comment by ph07** [comment]
http://docs.basho.com/dataplatform/1.0.0/learn-about-dataplatform/spark-cluster-manager-features/
http://docs.basho.com/dataplatform/1.0.0/learn-about-dataplatform/leader-election-service/

There are number of broken links in those articles.

**Comment by laurenrother** [comment]
@ph07 The AWS ports issue is being dealt with by Eric today, which we established yesterday. I forgot to mention it in my closing note. I closed the issue in an effort to discourage additions to it, as the comments are getting incredibly long and unwieldy at this point. New issues need to be opened on their own so that they may be dealt with in turn.

We do not have the ability to link to headers under size h2 at this time. The supported OS header attempts that. It does go to the correct page, and the h2 issue is being resolved through another project with our publishing platform. I'm therefore leaving that link as-is so that eventually it works wonderfully, while it currently works OK. 

We are also addressing the links-in-notes issue. I have Drew looking into whether that's a quick fix, and if it isn't, we're going to go through each page and change the note notation over. I forgot to mention that in my closing note.
